include/zephyr/posix/time_calls.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2022 Friedt Professional Engineering Services, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_POSIX_TIME_CALLS_H_
+#define ZEPHYR_INCLUDE_POSIX_TIME_CALLS_H_
+
+#include <time.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef CONFIG_POSIX_CLOCK
+__syscall int zephyr_clock_gettime(clockid_t clock_id, struct timespec *ts);
```

Comment:
If you want to elide the libc wrapper, you should be able to define clock_gettime directly as a 'gnu_inline' function (where supported, which means gcc or clang). That might be a nice optimization.
```
extern __inline int __attribute ((gnu_inline, always_inline)) clock_gettime(clock_id_t clock_id, struct timespec *ts) 
{
    return zephyr_clock_gettime(clock_id, ts);
}
```
that would allow the C library 'time.h' header to continue to declare clock_gettime as a regular function but all uses will be inlined to direct calls to the zephyr syscall version.

Reply:
Oh wow - that is quite a useful optimization. I worry though that there may be portability concerns.